src/components/app.jsx

```diff
@@ -541,6 +543,7 @@ const App = () => {
         setTabIndex(newTabIndex);
     }, []);
 
+    const urlMapper = useWebsocketUrlGenerator();
```

Comment:
I think we should keep the different sockets separated for now. Any code can call addListener/removeListener and refcount to open/close the websocket accordingly